src/overlay/TCPPeer.cpp

```diff
@@ -477,6 +475,10 @@ TCPPeer::startRead()
                 }
                 noteFullyReadBody(length);
                 recvMessage();
+                if (mApp.getClock().shouldYield())
```

Comment:
we should just move the code in the `else` block outside the while loop here as to avoid `break`: this makes the code harder to read